waymo2argo/waymo_raw_data_to_argoverse.py

```diff
@@ -0,0 +1,444 @@
+                "focal_length_y_px_": f_v,
+                "focal_center_x_px_": c_u,
+                "focal_center_y_px_": c_v,
+                "skew_": 0,
+                "distortion_coefficients_": [0, 0, 0],
+                "vehicle_SE3_camera_": {
+                    "rotation": {"coefficients": [qw, qx, qy, qz]},
+                    "translation": [x, y, z],
+                },
+            },
+        }
+        calib_dict["camera_data_"] += [cam_dict]
+    return calib_dict
+
+
+def dump_pose(
+    city_SE3_egovehicle: np.ndarray, timestamp: int, log_id: str, parent_path: str
+) -> None:
+    """Saves the SE3 transformation from city frame
+        to egovehicle frame at a particular timestamp
+
+    Args:
+        city_SE3_egovehicle: A (4,4) numpy array representing the
+                            SE3 transformation from city to egovehicle frame
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+    """
+    x, y, z = city_SE3_egovehicle[:3, 3]
+    R = city_SE3_egovehicle[:3, :3]
+    assert np.allclose(city_SE3_egovehicle[3], np.array([0, 0, 0, 1]))
+    q = rotmat2quat(R)
+    w, x, y, z = q
+    pose_dict = {"rotation": [w, x, y, z], "translation": [x, y, z]}
+    json_fpath = f"{parent_path}/{log_id}/poses/city_SE3_egovehicle_{timestamp}.json"
+    check_mkdir(str(Path(json_fpath).parent))
+    save_json_dict(json_fpath, pose_dict)
+
+
+def dump_point_cloud(
+    points: np.ndarray, timestamp: int, log_id: str, parent_path: str
+) -> None:
+    """Saves point cloud as .ply file extracted from Waymo's range images
+
+    Args:
+        points: A (N,3) numpy array representing the point cloud created from lidar readings
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+    """
+    data = {"x": points[:, 0], "y": points[:, 1], "z": points[:, 2]}
+    cloud = PyntCloud(pd.DataFrame(data))
+    cloud_fpath = f"{parent_path}/{log_id}/lidar/PC_{timestamp}.ply"
+    check_mkdir(str(Path(cloud_fpath).parent))
+    cloud.to_file(cloud_fpath)
+
+
+def dump_object_labels(
+    labels: List[waymo_open_dataset.label_pb2.Label],
+    timestamp: int,
+    log_id: str,
+    parent_path: str,
+    track_id_dict: Dict,
+) -> None:
+    """Saves object labels from Waymo dataset as json files
+
+    Args:
+        labels: A list of Waymo labels
+        timestamp: Timestamp in nanoseconds when the lidar reading occurred
+        log_id: Log ID that the reading belongs to
+        parent_path: The directory that the converted data is written to
+        track_id_dict: Dictionary to store object ID to track ID mappings
+    """
+    argoverse_labels = []
+    for label in labels:
+        if label.type != 3:
```

Comment:
thanks for updating this Hemanth. Can we make it `label.type != LABEL_TYPES.index("SIGN")` to make this explicit, and then a brief inline comment explaining we don't want SIGN, since not a category in Argoverse?

Reply:
Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/3b5f39632d27d0d05165fd850e73bd772ceaaaf5)